pymemcache/client/base.py

```diff
@@ -109,6 +109,27 @@ def check_key_helper(key, allow_unicode_keys, key_prefix=b''):
     return key
 
 
+def normalize_server_spec(server):
+    if isinstance(server, tuple) or server is None:
+        return server
+    if isinstance(server, list):
+        return tuple(server)  # Assume [host, port] provided.
+    if not isinstance(server, six.string_types):
+        raise ValueError('Unknown server specification provided.')
```

Comment:
Maybe include the `server` value and/or type in the exception message?

Reply:
Sure. Will output the repr.